src/lib/merkle-distributor/index.ts

```diff
@@ -23,17 +26,36 @@ import { MerkleDistributorInfo, parseBalanceMap } from './parse-balance-map';
  */
 export const createDistribution = (
   gifts: Record<string, number>,
+  fixedGifts: Record<string, BigNumber> | undefined,
```

Comment:
can we do this work around fixed/gift distinctions at a higher level of the code? the stuff in `lib/merkle-distributor` is intended to stay general

Reply:
Yeah, that makes sense. How about this?

```diff
diff --git a/src/lib/merkle-distributor/index.ts b/src/lib/merkle-distributor/index.ts
index f5e64088..5839f6a9 100644
--- a/src/lib/merkle-distributor/index.ts
+++ b/src/lib/merkle-distributor/index.ts
@@ -26,8 +26,16 @@ export const createDistribution = (
   totalAmount: BigNumber,
   previousDistribution?: Partial<MerkleDistributorInfo>
 ): MerkleDistributorInfo => {
+  const balances = balancesFromGifts(gifts, totalAmount);
+  return createDistributionFromBalances(balances, previousDistribution);
+};
+
+export const balancesFromGifts = (
+  gifts: Record<string, number>,
+  totalAmount: BigNumber
+) => {
   const totalGive = Object.values(gifts).reduce((t, v) => t + v);
-  let balances = Object.keys(gifts).map(address => ({
+  const balances = Object.keys(gifts).map(address => ({
     address,
     earnings: totalAmount.mul(gifts[address]).div(totalGive),
   }));
@@ -39,7 +47,13 @@ export const createDistribution = (
   const topBalance = assertDef(balances.find(x => x.address === topGift[0]));
   topBalance.earnings = topBalance.earnings.add(dust);
   assert(getDust(totalAmount, balances).eq(0), `dust is still not 0`);
+  return balances;
+};

+export const createDistributionFromBalances = (
+  balances: { address: string; earnings: BigNumber }[],
+  previousDistribution?: Partial<MerkleDistributorInfo>
+): MerkleDistributorInfo => {
   // add values from previous epoch
   if (previousDistribution) {
     assert(previousDistribution.claims, 'No claims found');

```

Then you could do this in useSubmitDistribution around line 84:

```js
const balances = balancesFromGifts(gifts, totalAmount);

// loop through balances and add fixed amounts

createDistributionFromBalances(balances, prev);
```

Reply:
Although-- you could also just leave it as-is if you want. I thought about it more and realized that the existing code is already not really that general, since it handles converting GIVE amounts based on a total.